src/shogun/classifier/LDA.cpp

```diff
@@ -95,45 +113,49 @@ bool CLDA::train_machine(CFeatures *data)
 			classidx_pos[num_pos++]=i;
 	}
 
-	w=SGVector<float64_t>(num_feat);
-	w.zero();
-	MatrixXd fmatrix=Map<MatrixXd>(feature_matrix.matrix, num_feat, num_vec);
-	VectorXd mean_neg(num_feat);
-	mean_neg=VectorXd::Zero(num_feat);
-	VectorXd mean_pos(num_feat);
-	mean_pos=VectorXd::Zero(num_feat);
+	SGVector<ST> w_st(num_feat);
```

Comment:
CLinearMachine contains a weights variable```SGVector<float64_t> w```.  I need a templated variable to store the weights during the training before I copy the final weight vector back into ```w``` ([line 265](https://github.com/shogun-toolbox/shogun/pull/3480/files#diff-7c03379ee9137c8e8719907bd06da7c2R265)).  If I were just to use ```SGVector<ST> w``` inside of the code, I'd get shadowing warnings.  So, I think it's better just to use ```w_st```.